From 3ffc85aedc606f0d3c328a5190cd829751c97a48 Mon Sep 17 00:00:00 2001
From: Moody Liu <mooodyhunter@outlook.com>
Date: Thu, 11 Jan 2024 02:00:49 +0000
Subject: [PATCH] libtaskmanager: do not leak fd when reading icons in
 waylandtaskmodel

BUG: 478831
FIXED-IN: 6.0


(cherry picked from commit 572aafa6eead5d1a56dfc5eefaccd7c76fb40a61)
---
 libtaskmanager/waylandtasksmodel.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libtaskmanager/waylandtasksmodel.cpp b/libtaskmanager/waylandtasksmodel.cpp
index 7825a8a9192..ec9d6f13b20 100644
--- a/libtaskmanager/waylandtasksmodel.cpp
+++ b/libtaskmanager/waylandtasksmodel.cpp
@@ -123,6 +123,9 @@ protected:
         get_icon(pipeFds[1]);
         ::close(pipeFds[1]);
         auto readIcon = [uuid = uuid](int fd) {
+            auto closeGuard = qScopeGuard([fd]() {
+                ::close(fd);
+            });
             pollfd pollFd;
             pollFd.fd = fd;
             pollFd.events = POLLIN;
-- 
GitLab

